padle/padle.py

```diff
@@ -632,15 +631,18 @@ async def set(self, ctx):
         """Set possible PADles to attached CSV/text file"""
         if ctx.message.attachments:
             try:
-                data = [int(x) for x in
-                        re.sub("[^0-9,]", "", (await ctx.message.attachments[0].read()).decode("utf-8")).split(",")]
+                input = StringIO((await ctx.message.attachments[0].read()).decode("utf-8"))
+                reader = csv.reader(input, delimiter=",")
+                valid = []
                 dbcog = await self.get_dbcog()
-                for id in data:
-                    m = dbcog.get_monster(int(id))
-                    if m is None or not m.on_na or m.name_en is None:
-                        return await send_cancellation_message(ctx, f"{id} was invalid. Please remove "
-                                                                    "this ID and try again.")
-                await self.config.monsters_list.set(data)
+                for row in reader:
+                    for id in row:
+                        m = dbcog.get_monster(int(id))
+                        if m is None or not m.on_na or m.name_en is None:
+                            return await send_cancellation_message(ctx, f"{id} was invalid. Please remove "
```

Comment:
can you build a complete list of incorrect ids as you go and return that all at once at the end instead of making them go through id by id